cpp/ql/test/library-tests/ir/ir/ir.cpp

```diff
@@ -988,6 +988,11 @@ int PointerDecay(int a[], int fn(float)) {
   return a[0] + fn(1.0);
 }
 
+int AsmStmt(int x) {
+  __asm__("");
+  return x;
+}
```

Comment:
Please add a non-trivial test as well, even if it's just to show that we don't translate it properly. Here is one you can lift: https://github.com/Semmle/ql/blob/c2a5350bf2/cpp/ql/test/query-tests/Likely%20Bugs/Arithmetic/PointlessComparison/RegressionTests.cpp#L9-L13.

I think we have reasonably good support for inline assembly, with `AsmStmt.getChild` returning each of the expressions that might get passed to the assembly statement. I don't know whether variables get passed by value or by reference.

Reply:
Oh, I didn't realize that `getChild()` had any results for `AsmStmt`. That means we should be able to model the side effects more precisely. I don't think that's blocking this PR, though. A Jira ticket to revisit this would be sufficient.

Reply:
I don't actually know if `AsmStmt.getChild` works on MSVC.